lhotse/recipes/mgb2.py

```diff
@@ -0,0 +1,203 @@
+    corpus_dir: Pathlike,
+    output_dir: Pathlike,
+    text_cleaning: bool = True,
+    buck_walter: bool = False,
+    num_jobs: int = 1,
+    mer_thresh:int = 80,
+) -> Dict[str, Dict[str, Union[RecordingSet, SupervisionSet]]]:
+
+    """
+    Returns the manifests which consist of the Recordings and Supervisions.
+    When all the manifests are available in the ``output_dir``, it will simply read and return them.
+
+    :param corpus_dir: Pathlike, the path of the data dir.
+    :param output_dir: Pathlike, the path where to write the manifests.
+    :param text_cleaning: Bool, if True, basic text cleaning is performed (similar to ESPNet recipe).
+    :param num_jobs: int, the number of jobs to use for parallel processing.
+    :return: a Dict whose key is the dataset part, and the value is Dicts with the keys 'audio' and 'supervisions'.
+
+    .. note::
+        Unlike other recipes, output_dir is not Optional here because we write the manifests
+        to the output directory while processing to avoid OOM issues, since it is a large dataset.
+
+    .. caution::
+        The `text_cleaning` option removes all punctuation and diacritics.
+    """
+    
+    corpus_dir = Path(corpus_dir)
+    assert corpus_dir.is_dir(), f"No such directory: {corpus_dir}"
+    
+    dataset_parts = ["dev", "train", "test"]
+    manifests = {}
+
+    if output_dir is not None:
+        output_dir = Path(output_dir)
+        output_dir.mkdir(parents=True, exist_ok=True)
+        # Maybe the manifests already exist: we can read them and save a bit of preparation time.
+        manifests = read_manifests_if_cached(
+            dataset_parts=dataset_parts, output_dir=output_dir, prefix="mgb2", suffix="jsonl.gz", lazy=True,
+        )
+
+    for part in dataset_parts:
+        logging.info(f"Processing MGB2 subset: {part}")
+        if manifests_exist(
+            part=part, output_dir=output_dir, prefix="mgb2", suffix="jsonl.gz"
+            ):
+            logging.info(f"MGB2 subset: {part} already prepared - skipping.")
+            continue
+
+        # Read the recordings and write them into manifest. We additionally store the
+        # duration of the recordings in a dict which will be used later to create the
+        # supervisions.
+        
+        if part == "test" or part == "dev":
+            os.system(f'mkdir -p {os.path.join(output_dir, part)}')
+            os.system(f"cp {os.path.join(corpus_dir, part,'text.non_overlap_speech')}\
+                 {os.path.join(output_dir, part,'text')}")
+            os.system(f"cp {os.path.join(corpus_dir, part,'segments.non_overlap_speech')}\
+                 {os.path.join(output_dir, part,'segments')}")
+            os.system(f"cat {os.path.join(corpus_dir, part,'wav.scp')}\
+                 | sed 's:wav/:{os.path.join(corpus_dir, part,'wav/')}:g' > {os.path.join(output_dir, part,'wav.scp')}") 
+
+            recordings, supervisions, _= load_kaldi_data_dir(os.path.join(output_dir, part), 16000)
```

Comment:
Does this mean that the Kaldi data preparation needs to be done before we can create the manifests? I think a data preparation recipe should be independent of this requirement.

Reply:
Yes, unfortunately the test and dev subsets are provided prepared with kaldi like style. I double checked with Dr Ahmed and he said that this is how MGB2 usually provided when someone downloads it.